### PR TITLE
Feature: SLR CIS data coverage generation script

### PR DESCRIPTION
**Pull request by clhenrick**
Adds a node js script that generates geojson files for the Sea Level Rise Coastal Inundation Scenarios tool's data coverage map. The script is added as an [npm workspace](https://docs.npmjs.com/cli/v7/using-npm/workspaces) and may be run from the root of the repository:

```bash
npm run slr-cis-data
```

### CODE REVIEW

**Review by mukhtyar — APPROVED** [review]
This is really interesting! Will need to dive into it a bit more to understand how npm workspaces function.

**Comment by clhenrick** [comment]
> This is really interesting! Will need to dive into it a bit more to understand how npm workspaces function.

thanks @mukhtyar, happy to go over it in our 1x1 wednesday.

I'm wondering if `static/data` is the best place to put the generated geojson files, or if they should go in a directory in `content` instead? I plan on fetching them async when generating the data coverage map in the UI so they don't need to bundled with the source code.

Also would you mind running the script to verify that it works? You'll need to do a `npm install` first.

File: scripts/slr-cis-data-coverage/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+This script generates the GeoJSON files for the Sea Level Rise – Coastal Inundation Scenarios tool's data coverage map.
+
+Run it from the root of this repository:
+
+```bash
+npm run slr-cis-data
+```
+
+GeoJSON files (one for each data source) will be output in `/static/data/`.
```
Comment (clhenrick) [inline scripts/slr-cis-data-coverage/README.md]: add: "plus one for calflod3d tfs 5m centroids"

File: scripts/slr-cis-data-coverage/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+**NOTE**: It's recommended to run this script on a device that uses an Linux or Linux like OS, not Windows.
```
Comment (clhenrick) [inline scripts/slr-cis-data-coverage/README.md]: change "not Windows" to "npm workspaces may have issues on Windows OS"

File: scripts/slr-cis-data-coverage/index.mjs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    if (name === "calflod3d_5m") {
+      // not all individual 5m tiles are in the API so we use a local geojson instead
+      return require("./calflod3d_5m_tile_index.json");
+    } else {
+      return await (
+        await fetch(`${API_BASE_URL}/?${LAYER_PARAMS.get(name)}&pagesize=125`)
+      ).json();
+    }
+  } catch (error) {
+    console.log(error);
```
Comment (clhenrick) [inline scripts/slr-cis-data-coverage/index.mjs]: use `handleError` here

**Comment by clhenrick** [comment]
Argh, noticing that in QGIS I should be using the dissolved calflod5m geojson for generating the centroids so that there's not redundancy:

image

but perhaps it's better to have the redundancy for seeing the individual tiles when zoomed out (zoom levels <= 6) calling it good for now.

**Comment by elehmer** [comment]
If you need me to do any of the unioning and dissolving let me know.

**Comment by clhenrick** [comment]
[SLR-CIS: data coverage locator map](https://trello.com/c/mkkYLpc8/148-slr-cis-data-coverage-locator-map)

**Comment by clhenrick** [comment]
> If you need me to do any of the unioning and dissolving let me know.

thanks @elehmer but I've already scripted it using [mapshaper](https://github.com/mbloch/mapshaper/wiki/Command-Reference) :)

**Pull request merged**